src/Gaffer/Metadata.cpp

```diff
@@ -810,40 +844,30 @@ IECore::ConstDataPtr Metadata::valueInternal( const GraphComponent *target, IECo
 	{
 		if( const Plug *plug = runTimeCast<const Plug>( target ) )
 		{
+			// Path-based values registered to a node take precedence over
+			// path-based values registered to a plug. This allows general
+			// metadata to be registered to descendants of a plug, but then
+			// overriden by more specific behaviour defined by metadata on
+			// the node using that plug.
+			if( const GraphComponent *node = plug->node() )
+			{
+				const vector<InternedString> matchPatternPath = StringAlgo::matchPatternPath( plug->relativeName( node ), '.' );
+				if( OptionalData result = plugPathMetadataValue( node->typeId(), plug, matchPatternPath, key ) )
+				{
+					return result.value();
+				}
+			}
```

Comment:
I think this is a bit too specific - I think _any_ higher-level registration should win, not just one on `plug->node()`. A couple of (slightly contrived) examples :

- A node might use a relative registration against plugs on an internal node with a known name.
- A TweakPlug-relative registration might override a Color3fPlug-relative registration for `TweakPlug.value.r`.

I think that can be achieved fairly cleanly by modifying the original loop slightly. Instead of returning as soon as we find a match, we would instead assign `vIt->second` to a variable, replacing any previous asssignment. Then when the loop completes, we call the function it contains.

Reply:
I've pushed an update that uses the suggested approach with an additional test to confirm the Node -> TweakPlug -> Color3fPlug overrides. 

As discussed offline, we found this change to introduce some issues with the TweakPlug `allowCreate` & `allowRemove` metadata not always registered at the right place relative to the tweak `mode` plug when its `presetNames` metadata was being forwarded though a Spreadsheet. We decided to use this as an opportunity to simplify things a bit and remove the `allowCreate` and `allowRemove` metadata entirely, as `allowCreate` was already registered to all Tweaks and it made sense to also allow the "Remove" mode everywhere. By adding a bit of forwarding to the `tweakPlugValueWidget:propertyType` metadata we can then stop copying around `tweakPlugValueWidget:*` metadata altogether...

This PR is still targetting `1.4_maintenance` in case we want to test a build internally, but this will need to target `main` and have the appropriate `Changes.md` entries added before merging.